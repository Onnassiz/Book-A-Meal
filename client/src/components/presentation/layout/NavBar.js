import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Logo from '../../../../assets/images/logo.png';

class NavBar extends Component {
	constructor(props) {
		super(props);
		this.signOut = this.signOut.bind(this);
	}

	signOut() {
		localStorage.removeItem('id');
		localStorage.removeItem('email');
		localStorage.removeItem('name');
		localStorage.removeItem('role');
		localStorage.removeItem('token');

		setTimeout(() => {
			window.location = '/';
		}, 500);
	}

	render() {
		const { user, page, cart } = this.props;
		// const initials = user.name.match(/\b(\w)/g).join('').toUpperCase();
		const initials = 'BA';
		return (
			<header>
				<h1 className="logo"><Link to="/"><img src={Logo} alt="logo" /></Link></h1>
				<div className="nav-left">
					<ul>
						{user.role === 'caterer' ?
							<li>
								<div className="dropdown">
									<a className={page === 'caterer' ? 'active' : ''}>{'Caterer\'s Links'}</a>
									<div className="dropdown-content">
										<Link to="/caterer/business_profile"><i className="material-icons">dashboard</i> Setup Profile</Link>
										<hr />
										<Link to="/caterer/meals"><i className="material-icons">donut_small</i> Manage Meals</Link>
										<Link to="/caterer/menus"><i className="material-icons">menu</i> Manage Menus</Link>
										<hr />
										<a href="pages/admin/manage-menu.html"><i className="material-icons">developer_board</i> View Reports</a>
									</div>
								</div>
							</li> : ''}
					</ul>
				</div>

				<div className="nav">
					<ul>
						<li className={page === 'menus' ? 'active' : ''}><Link to="/menus">Menu</Link></li>
						<li className={page === 'cart' ? 'active' : ''}><Link to="/cart" className={cart.cart.length ? 'dynamic-badge' : ''} data-badge={cart.cart.length}><i className="material-icons">shopping_cart</i> Cart</Link></li>
						<li><a href="#">Orders</a></li>
						<li>
							<div className="dropdown">
								<a href="#">{user.email}<i id="caret" className="material-icons">arrow_drop_down</i></a>
								<div className="dropdown-content">
									<div className="avatar-circle">
										<span className="initials">
											{initials}
										</span>
									</div>
									<hr />
									<a href="pages/admin/manage-meal.html"><i className="material-icons">settings</i> Profile</a>
									<a href="pages/admin/manage-menu.html"><i className="material-icons">developer_board</i> Change Password</a>
									<hr />
									<a onClick={this.signOut}><i className="material-icons">power_settings_new</i> Sign Out</a>
								</div>
							</div>
						</li>
					</ul>
				</div>
			</header>
		);
	}
}

NavBar.propTypes = {
	user: PropTypes.object.isRequired,
	page: PropTypes.string.isRequired,
};

export default NavBar;
