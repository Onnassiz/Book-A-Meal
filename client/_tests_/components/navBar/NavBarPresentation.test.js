import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavBar from '../../../src/components/presentation/layout/NavBar';
import userInfo from '../../objectProps/user.props';
import LocalStorageMock from '../../mocks/localStorage';

const props = {
  user: userInfo,
  page: 'home',
};

const emptyUserProps = {
  user: {},
  page: 'home',
};

global.localStorage = new LocalStorageMock();

const wrapper = shallow(<NavBar {...props} />);
const wrapperForEmptyUser = shallow(<NavBar {...emptyUserProps} />);

jest.useFakeTimers();
describe('<NavBar />', () => {
  it('Should render presentation component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should render presentation component when user is empty', () => {
    expect(toJson(wrapperForEmptyUser)).toMatchSnapshot();
  });

  it('Should sign delete user info from local storage when sign out is link is clicked', () => {
    wrapper.find('#signOut').simulate('click');
    jest.runAllTimers();
    expect(localStorage.getItem('id')).toEqual(null);
    expect(localStorage.getItem('email')).toEqual(null);
    expect(localStorage.getItem('role')).toEqual(null);
    expect(localStorage.getItem('name')).toEqual(null);
    expect(localStorage.getItem('token')).toEqual(null);
    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 500);
  });

  it('Should set show state when menu icon is clicked', () => {
    wrapper.find('#menuIcon').simulate('click');
    expect(wrapper.state().show).toBe('showAdmin');
  });

  it('Should set show state when menu icon is clicked and the current user is a customer', () => {
    const user = userInfo;
    user.role = 'customer';
    wrapper.setProps({ user });
    wrapper.setState({ show: '' });
    wrapper.find('#menuIcon').simulate('click');
    expect(wrapper.state().show).toBe('show');
  });
});
