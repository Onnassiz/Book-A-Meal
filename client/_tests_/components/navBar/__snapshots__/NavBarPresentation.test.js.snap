// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> Should render presentation component 1`] = `
<header>
  <h1
    className="logo"
  >
    <Link
      replace={false}
      to="/"
    >
      <img
        alt="logo"
        src="https://res.cloudinary.com/onnassiz/image/upload/v1535988132/logo_madffu.png"
      />
    </Link>
  </h1>
  <div
    className="nav-left"
  >
    <ul>
      <li>
        <div
          className="dropdown"
          id="caterer_dropdown"
        >
          <a
            className=""
          >
            Caterer's Workshop
            <i
              className="material-icons"
              id="caret"
            >
              arrow_drop_down
            </i>
          </a>
          <div
            className="dropdown-container"
          >
            <div
              className="dropdown-content"
            >
              <Link
                id="admin_profile"
                replace={false}
                to="/caterer/business_profile"
              >
                <i
                  className="material-icons"
                >
                  dashboard
                </i>
                 Setup Profile
              </Link>
              <hr />
              <Link
                id="admin_meals"
                replace={false}
                to="/caterer/meals"
              >
                <i
                  className="material-icons"
                >
                  donut_small
                </i>
                 Manage Meals
              </Link>
              <Link
                id="admin_menus"
                replace={false}
                to="/caterer/menus"
              >
                <i
                  className="material-icons"
                >
                  menu
                </i>
                 Manage Menus
              </Link>
              <hr />
              <a
                href="pages/admin/manage-menu.html"
              >
                <i
                  className="material-icons"
                >
                  developer_board
                </i>
                 View Reports
              </a>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="nav"
  >
    <ul>
      <li
        className=""
      >
        <Link
          className="main"
          id="user_menus"
          replace={false}
          to="/menus"
        >
          Menu
        </Link>
      </li>
      <li
        className=""
      >
        <Link
          className="main"
          id="user_orders"
          replace={false}
          to="/orders"
        >
          Orders
        </Link>
      </li>
      <li>
        <div
          className="dropdown"
          id="user_dropdown"
        >
          <button
            className="user-button"
          >
            <a>
              BO
              <i
                className="material-icons"
                id="caret"
              >
                arrow_drop_down
              </i>
            </a>
          </button>
          <div
            className="dropdown-container"
          >
            <div
              className="dropdown-content"
            >
              <div
                className="avatar-circle"
              >
                <span
                  className="initials"
                >
                  BO
                </span>
              </div>
              <hr />
              <Link
                replace={false}
                to="/profile"
              >
                <i
                  className="material-icons"
                >
                  settings
                </i>
                 Profile
              </Link>
              <hr />
              <a
                id="signOut"
                onClick={[Function]}
              >
                <i
                  className="material-icons"
                >
                  power_settings_new
                </i>
                 Sign Out
              </a>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="menuIcon"
  >
    <ul>
      <li>
        <button
          id="menuIcon"
          onClick={[Function]}
        >
          <i
            className="material-icons"
            style={
              Object {
                "fontSize": 30,
              }
            }
          >
            menu
          </i>
        </button>
      </li>
    </ul>
  </div>
  <div>
    <ul
      className="responsive "
    >
      <li
        className=""
      >
        <Link
          className="main"
          replace={false}
          to="/menus"
        >
          Menu
        </Link>
      </li>
      <li>
        <Link
          className="main"
          replace={false}
          to="/orders"
        >
          Orders
        </Link>
      </li>
      <hr />
      <a
        className="main"
        onClick={[Function]}
      >
        <i
          className="material-icons"
        >
          power_settings_new
        </i>
         Sign Out
      </a>
    </ul>
  </div>
</header>
`;

exports[`<NavBar /> Should render presentation component when user is empty 1`] = `
<header>
  <h1
    className="logo"
  >
    <Link
      replace={false}
      to="/"
    >
      <img
        alt="logo"
        src="https://res.cloudinary.com/onnassiz/image/upload/v1535988132/logo_madffu.png"
      />
    </Link>
  </h1>
  <div
    className="nav-left"
  >
    <ul />
  </div>
  <div
    className="nav"
  >
    <ul>
      <li>
        <Link
          id="register"
          replace={false}
          to="/auth/register"
        >
          <button
            className="auth-button"
          >
            Sign Up
          </button>
        </Link>
      </li>
      <li>
        <Link
          id="login"
          replace={false}
          to="/auth/login"
        >
          <button
            className="auth-button"
          >
            Login
          </button>
        </Link>
      </li>
    </ul>
  </div>
  <div
    className="menuIcon"
  >
    <ul>
      <li>
        <button
          id="menuIcon"
          onClick={[Function]}
        >
          <i
            className="material-icons"
            style={
              Object {
                "fontSize": 30,
              }
            }
          >
            menu
          </i>
        </button>
      </li>
    </ul>
  </div>
  <div>
    <ul
      className="responsive "
    >
      <li>
        <Link
          className="main"
          replace={false}
          to="/auth/login"
        >
          Sign In
        </Link>
      </li>
      <li>
        <Link
          className="main"
          replace={false}
          to="/auth/register"
        >
          Sign Up
        </Link>
      </li>
    </ul>
  </div>
</header>
`;
